showkase-component-impressions/src/main/java/com/vinaygaba/showkase_component_impressions/ComponentImpressions.kt

```diff
@@ -0,0 +1,145 @@
+package com.vinaygaba.showkase_component_impressions
+
+import android.annotation.SuppressLint
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.setValue
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.composed
+import androidx.compose.ui.layout.onGloballyPositioned
+import androidx.compose.ui.platform.LocalView
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.rememberUpdatedState
+import androidx.compose.ui.geometry.Rect
+import androidx.compose.ui.platform.LocalLifecycleOwner
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleEventObserver
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.flow.collect
+
+/**
+ * Passed in the callback of the [visibilityEvents] Modifier. Contains information about the
+ * visibility of a given composable function that uses the visibilityEvents Modifier.
+ */
+data class ShowkaseVisibilityEvent<T>(
```

Comment:
don't think a null key makes sense? also should you document what the key means/represents?
```suggestion
data class ShowkaseVisibilityEvent<T : Any>(
```